eng/tools/rush-runner.js

```diff
@@ -186,13 +191,23 @@ if (serviceDirs.length === 0) {
       }
       break;
 
-    default:
-      let requiredPackageNames = packageNames;
-      if (buildTransitiveDep) {
-        requiredPackageNames = getPackagesToBuild(packageNames, pkgGraph);
+    case "build":
+      if (actionComponents[1] === "samples") {
+        // For sample builds, we use --from to run sample builds on dependents
+        rushRunAll("--from", packageNames);
+      } else {
+        // For other builds, we use the transitive dependency logic if required, and build dependencies
+        // using --to
+        const requiredPackageNames = buildTransitiveDep
+          ? getPackagesToBuild(packageNames, pkgGraph)
+          : packageNames;
+
+        rushRunAll("--to", requiredPackageNames);
       }
-      params = flatMap(requiredPackageNames, (p) => [`--to`, p]);
-      spawnNode(baseDir, "common/scripts/install-run-rush.js", action, ...params, ...rushParams);
+      break;
+
+    default:
```

Comment:
Discussed the change to default behavior with @praveenkuttappan, and we think that the default case was changed as part of the transitive dependency work but that it probably should have remained the same, so we are reverting it here.